### DISCUSSION TITLE
Rendering or registering issue in multi-step form (Next.js)

### DISCUSSION THREAD

**Discussion by jetschni** [react-hook-form/react-hook-form#9892] [Q&A] [answered]
I am building a multi-step form in Next.js that consists of several steps, each with a single input field. The user can navigate between steps using next/previous buttons.

The form works as expected when each step has a different type of input field (e.g. text, textarea). You can see an example of this in [example 1](https://4l5kdi-3000.preview.csb.app/example1).

However, when two consecutive steps have the same type of input field (e.g. textarea), the text entered in the previous step is seen in the next step input field. In addition, this field only appear on submit in react-hook form `watchAllFields`. You can see an example of this in [example 2](https://4l5kdi-3000.preview.csb.app/example2). The problem only appear with input/textarea not with radio/select/checkbox.

I believe I may be doing something incorrect in regards to registering the input or rendering it properly. Any guidance or help how how to resolve this issue would be fantastic

The complete codesandbox project can be found here: [Codesandbox project](https://codesandbox.io/p/sandbox/react-hook-form-multi-step-4l5kdi?selection=%5B%7B%22endColumn%22%3A27%2C%22endLineNumber%22%3A4%2C%22startColumn%22%3A27%2C%22startLineNumber%22%3A4%7D%5D&file=%2Fcomponents%2FDynamicInterview.tsx)

**Comment by leapful** [comment, marked as answer]
@jetschni 

Actually, the issue is a common issue on React re-rendering, it happens even with `<input />` element, or for all cases that are using same element type on multiple steps.

The reason is because on re-rendering for same element type like `input` or `textarea`, React will try to re-use current DOM element to avoid real DOM change, therefore, the DOM element of step 1 will be used with new attribute like `name`, `id` ... for step 2 and the value of step 1 still persists.

Just add unique `key` prop to your `input` or `textarea` to notify that previous DOM element should be replaced.

Working demo:

https://codesandbox.io/p/sandbox/react-hook-form-multi-step-forked-tvekx8?selection=%5B%7B%22endColumn%22%3A31%2C%22endLineNumber%22%3A39%2C%22startColumn%22%3A31%2C%22startLineNumber%22%3A39%7D%5D&file=%2Fcomponents%2FControl.tsx

**Reply by jetschni** [reply]
Thank you very much @leapful . This is the solution, super simple and easy- fantastic!
